paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentSheetScreen.kt

```diff
@@ -460,11 +460,19 @@ private fun PrimaryButton(viewModel: BaseSheetViewModel) {
         modifier = modifier,
     )
 
-    LaunchedEffect(viewModel, button) {
+    LaunchedEffect(viewModel, button, currentScreen) {
         viewModel.primaryButtonUiState.collect { uiState ->
             withContext(Dispatchers.Main) {
                 button?.updateUiState(uiState)
             }
+            if (canEmitPMCompletedEvent && uiState?.enabled == true) {
+                viewModel.newPaymentSelection?.let { newPaymentSelection ->
+                    viewModel.eventReporter.onPaymentMethodFormCompleted(
```

Comment:
Thinking more about these, maybe we should follow the pattern set by `PaymentSheetAnalyticsListener` here. This "business logic" probably shouldn't live in the UI code.

Reply:
Actually, this seems like not the correct place at all to send this event. It seems like it should be coming from an Interactor, or somewhere deeper in the architecture.

Reply:
Maybe https://github.com/stripe/stripe-android/blob/428aab565363e8d40f2bb4c25cefaddf076dcde9/paymentsheet/src/main/java/com/stripe/android/paymentsheet/DefaultFormHelper.kt#L115